src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -267,9 +267,39 @@ private static XmlNode SetElement(XmlDocument document, XmlNode node, string nam
             SetAttribute(versionDetails, nodeToUpdate, VersionDetailsParser.NameAttributeName, itemToUpdate.Name);
             SetElement(versionDetails, nodeToUpdate, VersionDetailsParser.ShaElementName, itemToUpdate.Commit);
             SetElement(versionDetails, nodeToUpdate, VersionDetailsParser.UriElementName, itemToUpdate.RepoUri);
+        }
+    }
+
+    public async Task<GitFileContentContainer> UpdateDependencyFiles(
+        IEnumerable<DependencyDetail> itemsToUpdate,
+        SourceDependency sourceDependency,
+        string repoUri,
+        string branch,
+        IEnumerable<DependencyDetail> oldDependencies,
+        SemanticVersion incomingDotNetSdkVersion)
+    {
+        XmlDocument versionDetails = await ReadVersionDetailsXmlAsync(repoUri, branch);
+        XmlDocument versionProps = await ReadVersionPropsAsync(repoUri, branch);
+        JObject globalJson = await ReadGlobalJsonAsync(repoUri, branch);
+        JObject toolsConfigurationJson = await ReadDotNetToolsConfigJsonAsync(repoUri, branch);
+        XmlDocument nugetConfig = await ReadNugetConfigAsync(repoUri, branch);
+
+        foreach (DependencyDetail itemToUpdate in itemsToUpdate)
+        {
+            if (string.IsNullOrEmpty(itemToUpdate.Version) ||
```

Comment:
maybe write some kind CheckForNulls for the `DependencyDetail` class that will check these fields. We're doing the same thing at least one more time

Reply:
it could also throw the exception, since the messages are the same, maybe some `ThrowIfInvalid`

Reply:
Extracted into a `Validate()` method